packages/theme-language-server-common/src/completions/providers/ContentForBlockTypeCompletionProvider.ts

```diff
@@ -0,0 +1,42 @@
+import { NodeTypes } from '@shopify/liquid-html-parser';
+import { CompletionItem, CompletionItemKind } from 'vscode-languageserver';
+import { LiquidCompletionParams } from '../params';
+import { Provider } from './common';
+
+export class ContentForBlockTypeCompletionProvider implements Provider {
+  constructor(
+    private readonly getThemeBlockNames: (
+      rootUri: string,
+      includePrivate: boolean,
+    ) => Promise<string[]>,
+  ) {}
+
+  async completions(params: LiquidCompletionParams): Promise<CompletionItem[]> {
+    if (!params.completionContext) return [];
+
+    const { document } = params;
+    const doc = document.textDocument;
+    const { node, ancestors } = params.completionContext;
+    const parentNode = ancestors.at(-1);
+    const grandParentNode = ancestors.at(-2);
+
+    if (
+      !node ||
+      !parentNode ||
+      !grandParentNode ||
+      node.type !== NodeTypes.String ||
+      parentNode.type !== NodeTypes.NamedArgument ||
+      parentNode.name !== 'type' ||
+      grandParentNode.type !== NodeTypes.ContentForMarkup ||
+      grandParentNode.contentForType.value !== 'block'
+    ) {
+      return [];
+    }
+
+    return (await this.getThemeBlockNames(doc.uri, false)).map((blockName) => ({
+      label: blockName,
+      kind: CompletionItemKind.Keyword,
```

Comment:
It doesn't seem like we're completing a keyword, so I personally believe we should use `CompletionItemKind.EnumMember`, as the completion items are part of a well-defined set of options (standardized and formally grouped).

Alternatively, we could use `CompletionItemKind.Value` as well, but it doesn't seem so ideal since it refers to literal values not tied to a certain group.

(Just mentioning this because this impacts the icons of the completion items, so it also gives extra clues for users) :)